addons/web/static/src/views/basic_relational_model.js

```diff
@@ -1351,7 +1351,11 @@ export class RelationalModel extends Model {
             const legacyOptions = mapDoActionOptionAPI(payload.options);
             return this.actionService.doAction(payload.action, legacyOptions);
         } else if (evType === "reload") {
-            return this.load();
+            return this.load().then(() => {
+                if (ev.data.onSuccess) {
+                    ev.data.onSuccess();
```

Comment:
since we now wait for the confirm callback's promise to be resolved before closing, a form view test failed and highlighted another bug: by not calling onSuccess, we did not close the confirmation dialog after archiving a record